src/core/taylor.rkt

```diff
@@ -175,6 +173,36 @@
      (taylor-pow (normalize-series (taylor var base)) power)]
     [`(pow ,base ,power)
      (taylor var `(exp (* ,power (log ,base))))]
+    [`(expm1 ,arg) (taylor var `(- (exp ,arg) 1))]
+    [`(log1p ,arg) (taylor var `(log (+ 1 ,arg)))]
+    [`(hypot ,a ,b)
+     (define ta (taylor var a))
+     (define tb (taylor var b))
+     (taylor-sqrt (taylor-add ta tb))]
+    [`(fma ,a ,b ,c) (taylor var `(+ (* ,a ,b) ,c))]
+    [`(sinh ,arg)
+     (define exparg (taylor var `(exp ,arg)))
+     (taylor-mult (taylor-exact 1/2) (taylor-add exparg (taylor-negate (taylor-invert exparg))))]
+    [`(cosh ,arg)
+     (define exparg (taylor var `(exp ,arg)))
+     (taylor-mult (taylor-exact 1/2) (taylor-add exparg (taylor-invert exparg)))]
+    [`(tanh ,arg)
+     (define exparg (taylor var `(exp ,arg)))
+     (define expinv (taylor-invert exparg))
+     (define x+ (taylor-add exparg expinv))
+     (define x- (taylor-add exparg (taylor-negate expinv)))
+     (taylor-quotient x- x+)]
+    [`(asinh ,x)
+     (define tx (taylor var x))
+     (taylor-log (taylor-add tx (taylor-sqrt (taylor-add (taylor-mult tx tx) (taylor-exact 1)))))]
+    [`(acosh ,x)
+     (define tx (taylor var x))
+     (taylor-log (taylor-add tx (taylor-sqrt (taylor-add (taylor-mult tx tx) (taylor-exact -1)))))]
+    [`(atanh ,x)
+     (define tx (taylor var x))
+     (taylor-mult (taylor-exact 1/2)
+                  (taylor-log (taylor-quotient (taylor-add (taylor-exact 1) tx)
+                                               (taylor-add (taylor-exact 1) (taylor-negate tx)))))]
```

Comment:
Whoa. These look exciting. Not sure how many benchmarks this even applies to though.

Reply:
It's not as exciting as it looks. These were previously done by `function-definitions.rkt` which I found to be near-unfixable and decided to just rip out. The versions here are a little faster because they share subterms but it shouldn't affect the results any.